src/algebra/associated.lean

```diff
@@ -163,7 +163,7 @@ begin
   exact H _ o.1 _ o.2 h.symm
 end
 
-lemma irreducible_of_prime [comm_cancel_monoid_with_zero α] {p : α} (hp : prime p) :
+lemma prime.irreducible [comm_cancel_monoid_with_zero α] {p : α} (hp : prime p) :
```

Comment:
```suggestion
protected lemma prime.irreducible [comm_cancel_monoid_with_zero α] {p : α} (hp : prime p) :
```
Probably necessary to avoid weird errors!